### PR TITLE
Tcl: Create basic language file for tcl

### PR DESCRIPTION
**Pull request by pareksha**
This creates a basic language file for tcl in
coala language definitions.

Closes https://github.com/coala/coala/issues/5570

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: coalib/bearlib/languages/definitions/Tcl.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class Tcl:
+    aliases = 'Tcl/Tk',
+    versions = 8.0, 8.1, 8.2, 8.3, 8.4, 8.5, 8.6
```
Comment (jayvdb) [inline coalib/bearlib/languages/definitions/Tcl.py]: as with m4, I dont feel we're in a good position to define language versions at this stage.
In this case, Tcl the program and Tcl the language were quite closely tied together, but we dont know which program releases included language syntax changes.

File: coalib/bearlib/languages/definitions/Tcl.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    extensions = '.tcl',
+    comment_delimiter = '#'
+    string_delimiters = {'"': '"'}
+    multiline_string_delimiters = {'"': '"'}
+    encapsulators = {'(': ')', '[': ']', '{': '}', '[[': ']]'}
```
Comment (jayvdb) [inline coalib/bearlib/languages/definitions/Tcl.py]: what is the meaning of `[[ ... ]]` ?

File: coalib/bearlib/languages/definitions/Tcl.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from coalib.bearlib.languages.Language import Language
+
+
+@Language
+class Tcl:
+    aliases = 'Tcl/Tk',
+    versions = 8.0, 8.1, 8.2, 8.3, 8.4, 8.5, 8.6
+    extensions = '.tcl',
+    comment_delimiter = '#'
```
Comment (jayvdb) [inline coalib/bearlib/languages/definitions/Tcl.py]: not quite..

'#`' only works at the beginning of a command, which is conceptually incompatible with the other usages of `comment_delimiter` which assume a comment delimiter has a high precedence.

Typically this means in Tcl, `; #` is needed to be a comment occurring in the middle of a line.

This difference is quite small, and quite achievable to support, but is getting further from your project scope of quotes, so I think discard this from the MR, and please create an issue about supporting Tcl comments.

File: coalib/bearlib/languages/definitions/Tcl.py (hunk 2 above)
Comment (pareksha) [reply coalib/bearlib/languages/definitions/Tcl.py]: oops, mistakenly took nested ones as encapsulators, will remove this.

File: coalib/bearlib/languages/definitions/Tcl.py (hunk 1 above)
Comment (pareksha) [reply coalib/bearlib/languages/definitions/Tcl.py]: Right, I better remove versions

File: coalib/bearlib/languages/definitions/Tcl.py (hunk 3 above)
Comment (pareksha) [reply coalib/bearlib/languages/definitions/Tcl.py]: I have created an issue regarding this here https://github.com/coala/coala/issues/5580. 

For now, I have removed `comment_delimiter` from this PR.

**Comment by jayvdb** [comment]
ack 5bb4953

**Comment by jayvdb** [comment]
@gitmate-bot rebase

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**